src/suites/cts/validation/createBindGroup.spec.ts

```diff
@@ -0,0 +1,491 @@
+    return errorBuffer;
+  }
+}
+
+export const g = new TestGroup(F);
+
+g.test('binding count mismatch', async t => {
+  const bindGroupLayout = t.device.createBindGroupLayout({
+    bindings: [
+      {
+        binding: 0,
+        visibility: GPUShaderStage.COMPUTE,
+        type: 'storage-buffer',
+      },
+    ],
+  });
+
+  const goodDescriptor: GPUBindGroupDescriptor = {
+    bindings: [
+      {
+        binding: 0,
+        resource: {
+          buffer: t.getStorageBuffer(),
+        },
+      },
+    ],
+    layout: bindGroupLayout,
+  };
+
+  const badDescriptor: GPUBindGroupDescriptor = {
+    bindings: [
+      {
+        binding: 0,
+        resource: {
+          buffer: t.getStorageBuffer(),
+        },
+      },
+      // Another binding is added.
+      {
+        binding: 1,
+        resource: {
+          buffer: t.getStorageBuffer(),
+        },
+      },
+    ],
+    layout: bindGroupLayout,
+  };
+
+  // Control case
+  t.device.createBindGroup(goodDescriptor);
+
+  // Another binding is not expected.
+  await t.expectValidationError(() => {
+    t.device.createBindGroup(badDescriptor);
+  });
+});
+
+g.test('binding must be present in layout', async t => {
+  const bindGroupLayout = t.device.createBindGroupLayout({
+    bindings: [
+      {
+        binding: 0,
+        visibility: GPUShaderStage.COMPUTE,
+        type: 'storage-buffer',
+      },
+    ],
+  });
+
+  const goodDescriptor: GPUBindGroupDescriptor = {
+    bindings: [
+      {
+        binding: 0,
+        resource: {
+          buffer: t.getStorageBuffer(),
+        },
+      },
+    ],
+    layout: bindGroupLayout,
+  };
+
+  const badDescriptor: GPUBindGroupDescriptor = {
+    bindings: [
+      {
+        binding: 1, // binding index becomes 1.
+        resource: {
+          buffer: t.getStorageBuffer(),
+        },
+      },
+    ],
+    layout: bindGroupLayout,
+  };
+
+  // Control case
+  t.device.createBindGroup(goodDescriptor);
+
+  // Binding index 0 must be present.
+  await t.expectValidationError(() => {
+    t.device.createBindGroup(badDescriptor);
+  });
+});
+
+g.test('buffer binding must contain exactly one buffer of its type', async t => {
+  const { type } = t.params;
+
+  let matchedResource: GPUBindingResource;
+  if (type === 'uniform-buffer') {
+    matchedResource = { buffer: t.getUniformBuffer() };
+  } else if (type === 'storage-buffer' || type === 'readonly-storage-buffer') {
+    matchedResource = { buffer: t.getStorageBuffer() };
+  } else if (type === 'sampler') {
+    matchedResource = t.getSampler();
+  } else if (type === 'sampled-texture') {
+    matchedResource = t.getSampledTexture().createView();
+  } else if (type === 'storage-texture') {
+    matchedResource = t.getStorageTexture();
+  } else {
+    throw new Error('Unexpected binding type');
+  }
+
+  async function* mismatchedResources(): AsyncIterable<GPUBindingResource> {
+    const errorBuffer = await t.getErrorBuffer();
+    yield { buffer: errorBuffer };
```

Comment:
Well, unfortunately I had to remove it here, in order to split the test into more cases. Because we don't implement storage-texture yet, it was causing all of the tests to fail and losing test coverage on everything else.